satellite/analytics/service.go

```diff
@@ -132,10 +222,59 @@ func (service *Service) TrackCreateUser(fields TrackCreateUserFields) {
 		Event:       eventAccountCreated,
 		Properties:  props,
 	})
+
+	fullName := fields.FullName
+	names := strings.SplitN(fullName, " ", 2)
+
+	var firstName string
+	var lastName string
+
+	if len(names) > 1 {
+		firstName = names[0]
+		lastName = names[1]
+	} else {
+		firstName = fullName
+	}
+
+	data := PayloadUser{
+		Email: fields.Email,
+		Properties: PropertiesUser{
+			Email:          fields.Email,
+			Firstname:      firstName,
+			Lastname:       lastName,
+			Lifecyclestage: "customer",
+		},
+	}
+
+	dataEvent := map[string]interface{}{
+		"email":     fields.Email,
+		"eventName": "pe20293085_account_created_new",
+		"properties": map[string]interface{}{
+			"userid":             fields.ID.String(),
+			"email":              fields.Email,
+			"name":               fields.FullName,
+			"satellite_selected": service.satelliteName,
+			"account_type":       string(fields.Type),
+			"company_size":       fields.EmployeeCount,
+			"company_name":       fields.CompanyName,
+			"job_title":          fields.JobTitle,
+			"have_sales_contact": fields.HaveSalesContact,
+		},
+	}
+
+	err := service.createHubSpotUser(ctx, data)
```

Comment:
@mobyvb note, there must not be any blocking call to the tracking server. There's no need to slow down regular user interactions due to tracking.

Reply:
@egonelbre which of these would you prefer?
```
go func() {
    service.analytics.TrackCreateUser()
}()
```
(caller side)

or 

```
func (service *Service) TrackCreateUser(fields TrackCreateUserFields) {
    go func() {
        ...
    }()
}
```
(callee side)

or something else?

Reply:
We could also just add the goroutine around the calls to the new methods that Prerna has added. I think the segment events are already sent async (I need to verify though)

Reply:
See `enqueueMessage`.